### PR TITLE
Updating samples to v1 version

### PR DESCRIPTION
**Pull request by soltysh**
I'm currently updating all the json samples to work with v1 version of our API.

// cc @bparees @mfojtik @danmcp @adellape

### CODE REVIEW

File: dev_guide/integrating_external_services.adoc [hunk 1]
```diff
@@ -53,12 +53,20 @@ ignore the service and allow you to specify endpoints manually:
 ----
 {
   "kind": "Service",
-  "apiVersion": "v1beta3",
+  "apiVersion": "v1",
   "metadata": {
     "name": "example-mysql-service"
   },
-  "spec": { <1>
```
Comment (bparees) [inline dev_guide/integrating_external_services.adoc]: you removed the link to the footnote.  though i'm not sure what that footnote is trying to say anyway.

File: dev_guide/integrating_external_services.adoc [hunk 2]
```diff
@@ -77,11 +85,11 @@ the location to send traffic going to the service.
 ----
 {
   "kind": "Endpoints",
-  "apiVersion": "v1beta3",
+  "apiVersion": "v1",
   "metadata": {
-    "name": "external-mysql-service" <1>
```
Comment (bparees) [inline dev_guide/integrating_external_services.adoc]: why did you remove the footnote link?

File: dev_guide/integrating_external_services.adoc [hunk 3]
```diff
@@ -77,11 +85,11 @@ the location to send traffic going to the service.
...
+    "name": "external-mysql-service"
   },
-  "endpoints": [ <2>
```
Comment (bparees) [inline dev_guide/integrating_external_services.adoc]: again

File: dev_guide/integrating_external_services.adoc [hunk 4]
```diff
@@ -100,63 +108,70 @@ service by setting environment variables in the appropriate containers. In the f
...
 
 ----
 {
   "kind": "DeploymentConfig",
-  "apiVersion": "v1beta1",
+  "apiVersion": "v1",
   "metadata": {
     "name": "my-app-deployment"
   },
-  "template": { <1>
```
Comment (bparees) [inline dev_guide/integrating_external_services.adoc]: footnote

**Comment by adellape** [comment]
Wowee, :bow:.

**Comment by soltysh** [comment]
@bparees yeah I know I need to review those changes carefully because of
those footnotes. I was already tired yesterday and I wanted to create this
PR to let everyone know. Still thx for the early review :-)

File: dev_guide/integrating_external_services.adoc (hunk 1 above)
Comment (soltysh) [inline dev_guide/integrating_external_services.adoc]: I think there should be added:
```
        "selector": {
        }
```
to which the footnote should be referring to, but I'm not sure if it makes sense. @mfojtik your knowledge is needed in this topic, would adding empty selector match what the footnote is saying?

Comment (bparees) [inline dev_guide/integrating_external_services.adoc]: yeah that's what it was supposed to link to and if you read the paragraph above (which says "leave the selector field unset") then it makes sense i guess.

Comment (soltysh) [inline dev_guide/integrating_external_services.adoc]: Yeah, I'm not an expert in that area, but @mfojtik was dealing with them, so I'm hoping to get his answer on this. Clearly that was an error in the docs.

**Comment by soltysh** [comment]
I think I'm done, though there are a couple of places still remaining update to v1, but I just didn't know how to update them properly as of now. These places are:
```
$ ack "v1beta[13]"
dev_guide/deployments.adoc
298:            "version": "v1beta1",

admin_guide/selfprovisioned_projects.adoc
31:apiVersion: v1beta3
37:- apiVersion: v1beta1
48:- apiVersion: v1beta1

admin_guide/master_node_configuration.adoc
115:- v1beta3
172:  - v1beta3

architecture/core_objects/kubernetes_model.adoc
186:    "selfLink": "/api/v1beta1/replicationControllers",
188:    "apiVersion": "v1beta1",
194:            "selfLink": "/api/v1beta1/replicationControllers/docker-registry-1?namespace=default",
```
I'd like to ask authors of these files (@ironcladlou, @deads2k) to update those, don't forget about selfLinks :wink: 

@adellape @bparees @danmcp @mfojtik can I ask you guys for detailed review, I don't want to mess something up. I've made sure that each of those examples works so that part should be ok. But I might remove footnote links like previously @bparees noted or alike.

File: admin_guide/examples/geo-cache.json [hunk 5]
```diff
@@ -1,55 +1,74 @@
...
-                "ha-cache": "geo"
-            }
+            "targetPort": 0,
+            "nodePort": 0
+          }
+        ],
+        "selector": {
+          "ha-cache": "geo"
+        },
+        "portalIP": "",
```
Comment (bparees) [inline admin_guide/examples/geo-cache.json]: you can just remove this.

File: admin_guide/examples/geo-cache.json [hunk 6]
```diff
@@ -1,55 +1,74 @@
...
+        "type": "ClusterIP",
+        "sessionAffinity": "None"
+      },
+      "status": {
```
Comment (bparees) [inline admin_guide/examples/geo-cache.json]: delete?

File: admin_guide/examples/geo-cache.json [hunk 7]
```diff
@@ -1,55 +1,74 @@
...
+                    "protocol": "TCP"
+                  }
+                ]
+              }
+            ]
+          }
         }
-    ]
+      },
+      "status": {
```
Comment (bparees) [inline admin_guide/examples/geo-cache.json]: delete?

File: admin_guide/manage_nodes.adoc [hunk 8]
```diff
@@ -98,21 +98,26 @@ scheduled for the node.
...
+  "kind": "Node", <1>
+  "apiVersion": "v1", <2>
+  "metadata": {
+    "name": "node3.example.com", <3>
+    "labels": { <4>
+      "name": "node3"
+    }
   },
+  "spec": {
+    "externalID": "node3.example.com"
```
Comment (bparees) [inline admin_guide/manage_nodes.adoc]: footnote this?

File: architecture/core_objects/kubernetes_model.adoc [hunk 9]
```diff
@@ -243,35 +240,38 @@ attached:
...
+    ],
+    "selector": {
+      "name": "frontend" <4>
     },
-    "status": {}
+    "portalIP": "172.30.217.106", <5>
+    "type": "ClusterIP",
+    "sessionAffinity": "None"
+  },
+  "status": {
```
Comment (bparees) [inline architecture/core_objects/kubernetes_model.adoc]: delete

File: architecture/core_objects/openshift_model.adoc [hunk 10]
```diff
@@ -448,31 +438,33 @@ annotations.
...
+      "displayname": "Hello OpenShift"
     }
+  },
+  "spec": {
+    "finalizers": [
+      "openshift.com/project",
+      "kubernetes"
+    ]
+  },
+  "status": {
```
Comment (bparees) [inline architecture/core_objects/openshift_model.adoc]: delete?  (nothing we are illustrating as sample for a user to create from should have status i don't think.. we should only see status in samples of what is retrieved from the system)

File: creating_images/sti_testing.adoc [hunk 11]
```diff
@@ -143,45 +143,54 @@ a new STI builder image.
...
   },
-  "triggers": [
-    {
-      "type": "github",
-      "github": {
-        "secret": "secret101"
+  "spec": {
+    "triggers": [
+      {
+        "type": "github",
```
Comment (bparees) [inline creating_images/sti_testing.adoc]: GitHub

File: creating_images/sti_testing.adoc [hunk 12]
```diff
@@ -143,45 +143,54 @@ a new STI builder image.
...
       }
     },
     "strategy": {
       "type": "Custom",
       "customStrategy": {
-        "image": "openshift/sti-image-builder",
-        "exposeDockerSocket": true,
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "sti-image-builder:latest"
```
Comment (bparees) [inline creating_images/sti_testing.adoc]: this won't work unless you also have them setup the imagestream.

or you can change the Kind to DockerImage and make the name be "openshift/sti-image-builder" again.

File: dev_guide/builds.adoc [hunk 13]
```diff
@@ -366,13 +370,14 @@ trigger on:
...
-    "name": "ruby-20-centos7",
-  },
   "kind": "ImageStream",
-  "apiVersion": "v1beta1",
-}
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "ruby-20-centos7"
+  },
+  "status": {
```
Comment (bparees) [inline dev_guide/builds.adoc]: delete

File: dev_guide/builds.adoc [hunk 14]
```diff
@@ -366,13 +370,14 @@ trigger on:
...
+    "dockerImageRepository": ""
+  }
```
Comment (bparees) [inline dev_guide/builds.adoc]: this example is broken...  (it was broken even before you touched it).  this ImageStream doesn't actually point anywhere (no docker repository).  looks like it was supposed to point to ruby-20-centos7.

Comment (bparees) [inline dev_guide/builds.adoc]: ooops, nevermind. it's going to use the default internal registry and the metadata.name.

File: dev_guide/deployments.adoc [hunk 15]
```diff
@@ -34,59 +34,60 @@ new deployment every time a Docker image tag changes:
 
 ====
 
-[source,json]
+[source.json]
```
Comment (bparees) [inline dev_guide/deployments.adoc]: in other places you made the periods into commas, here you made a comma into a period.....?

**Comment by bparees** [comment]
@soltysh comments made.

File: creating_images/sti_testing.adoc (hunk 12 above)
Comment (soltysh) [inline creating_images/sti_testing.adoc]: I think we want to advertise our ImageStreams here as much as possible, the example as is is working just fine and getting information that you're missing some IS is perfectly ok when someone tries to use this as an example, imho.

File: admin_guide/manage_nodes.adoc (hunk 8 above)
Comment (soltysh) [inline admin_guide/manage_nodes.adoc]: This is completely removed in the new docs.

**Comment by soltysh** [comment]
Rebased, @bparees addressed your comments with 2 exceptions (one about footnode for node - that part is gone in latest docs, two - replied inline). 
@adellape @bparees appreciate your final review and if you're ok I'd like to get it merged as this touches quite a lot and is conflicting very frequently ;)

File: creating_images/sti_testing.adoc (hunk 12 above)
Comment (bparees) [inline creating_images/sti_testing.adoc]: in general I'd agree, on this one i'm less sure I agree.  we're not going to create an openshift/sti-image-builder stream out of the box, it's not really a normal thing to need/want.

Comment (soltysh) [inline creating_images/sti_testing.adoc]: Right..., I didn't take this into account. I'll update to use DockerImage.

**Comment by soltysh** [comment]
@bparees updated.

File: architecture/core_objects/openshift_model.adoc [hunk 16]
```diff
@@ -67,84 +67,87 @@ OpenShift are immutable.
 
 Like Docker, images are retrieved within the scope of an image stream. The sample image below is from the `origin-ruby-sample` image stream, and can be retrieved by asking for the `ImageStreamImage` with name `origin-ruby-sample@sha256:4d3a646b58685449179a0c61ad4baa19a8df8ba668e0f0704b9ad16f5e16e642`.
 
+====
```
Comment (adellape) [inline architecture/core_objects/openshift_model.adoc]: It looks like the opening example blocks being introduced throughout this topic are missing their closing `====`s, which is messing up any section titles that follow.

File: dev_guide/builds.adoc [hunk 17]
```diff
@@ -26,7 +26,9 @@ A buildConfig is a REST object, which can be used in a POST to the API server to
 create a new instance. The following configuration results in a new build
 every time a Docker image tag or the source code changes:
 
-[source.json]
+====
```
Comment (adellape) [inline dev_guide/builds.adoc]: Similar issue in this topic w/ a few `====` blocks.

**Comment by adellape** [comment]
@soltysh LGTM other than the handful of breaking example blocks.

**Comment by soltysh** [comment]
@adellape fixed, hopefully all of them. Final review and merge :facepunch:

**Comment by adellape** [comment]
@soltysh LGTM, thanks! :clap: Will merge after build passes.

**Comment by soltysh** [comment]
:cocktail:
